unity/Holo/Assets/Scripts/ModelClippingPlaneControl.cs

```diff
@@ -0,0 +1,142 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using HoloToolkit.Unity.Buttons;
+using HoloToolkit.Unity.InputModule;
+using HoloToolkit.Unity.UX;
+
+public class ModelClippingPlaneControl : MonoBehaviour, IClickHandler
+{
+    public CompoundButton ClippingPlaneBtn;
+    public CompoundButton ClippingPlaneTranslationBtn;
+    public CompoundButton ClippingPlaneRotationBtn;
+    public Material ClippingMaterial;
+    public Material DefaultModelMaterial;
+
+    public GameObject LoadedModel;
+
+    BoundingBoxRig clipPlaneQuadBbox;
+
+    private HandDraggable HandTranslation;
+    
+    
+    private enum ClipPlaneState
+    {
+        Disabled,
+        Active,
+        Translation,
+        Rotation
+    }
+
+    ClipPlaneState ClippingPlaneState = ClipPlaneState.Disabled;
+
+    void Start()
+    {
+        clipPlaneQuadBbox = GetComponent<BoundingBoxRig>();
+        HandTranslation = GetComponent<HandDraggable>();
+        HandTranslation.enabled = false;
+
+        ResetState();
+    }
+
+    public void Click(GameObject clickObj)
+    {
+        Debug.Log("Clicked obj: " + clickObj.name);
+        switch (clickObj.name)
+        {
+            case "ButtonClipping":
+                ClickedClipPlane();
+                break;
+
+            case "ButtonClippingTranslation":
+                ClickedClipPlaneTranslation();
+                break;
+
+            case "ButtonClippingRotation":
+                ClickedClipPlaneRotation();
+                break;
+        }
+
+    }
+
+    void ClickedClipPlaneTranslation()

```

Comment:
I am *not* sure is this the right suggestion, but I would try to merge ClickedClipPlaneTranslation and ClickedClipPlaneRotation into one, and make it look a bit more like `ClickChangeTransformationState` https://github.com/MicroscopeIT/holo/blob/master/unity/Holo/Assets/Scripts/model_with_plate/ModelWithPlate.cs#L316 . This way:

- The logic that "if you click on a button indicating the current state (you click on rotation while rotation is active etc.) -> then you switch to ClipPlaneState.Active" is implemented in one place (right now, both ClickedClipPlaneRotation and ClickedClipPlaneTranslation need to account for this, using slightly different code). Like this:

```
if (newState == ClippingPlaneState && newState != ClipPlaneState.Disabled) {
  // clicking again on rotation/translation button turns the "transformation mode" off
  newState = ClipPlaneState.Active;
}
```

- You could write one assignment `HandTranslation.enabled = newState == ClipPlaneState.Translation;`, thus avoiding the current 3 assignments to `HandTranslation.enabled` in ClickedClipPlaneRotation and ClickedClipPlaneTranslation

- You could write 
```
ModelWithPlate.SetButtonState(ClippingPlaneTranslationBtn, ClippingPlaneState == ClipPlaneState.Translation);
ModelWithPlate.SetButtonState(ClippingPlaneRotationBtn, ClippingPlaneState == ClipPlaneState.Rotation);
```
which looks clear (i.e., it's immediately obvious that at most one button will be in "active" state).

Reply:
This is good suggestion :) I've merged them thanks.